src/spatialdata/_io/io_zarr.py

```diff
@@ -12,62 +13,73 @@
 from spatialdata._io.io_points import _read_points
 from spatialdata._io.io_raster import _read_multiscale
 from spatialdata._io.io_shapes import _read_shapes
+from spatialdata._logging import logger
 from spatialdata.models import TableModel
 
 
 def read_zarr(store: Union[str, Path, zarr.Group]) -> SpatialData:
-    if isinstance(store, str):
-        store = Path(store)
+    f = store if isinstance(store, zarr.Group) else zarr.open(store, mode="r")
 
-    f = zarr.open(store, mode="r")
     images = {}
     labels = {}
     points = {}
     table: Optional[AnnData] = None
     shapes = {}
 
     # read multiscale images
-    images_store = store / "images"
-    if images_store.exists():
-        f = zarr.open(images_store, mode="r")
-        for k in f:
-            f_elem = f[k].name
-            f_elem_store = f"{images_store}{f_elem}"
-            images[k] = _read_multiscale(f_elem_store, raster_type="image")
+    if "images" in f:
+        group = f["images"]
+        count = 0
+        for k in group:
```

Comment:
Minor nitpick: I would prefer if the iterator variable had a more clear name (e.g., `subgroup_name` instead of `k`).